src/plugins/data/public/interpreter/setup.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { FunctionsRegistry, TypesRegistry } from '../../common/interpreter/registries';
+import { InterpreterSetup } from './types';
+import { RenderFunctionsRegistry } from './util/render_function';
+
+export const setup = (): InterpreterSetup => {
```

Comment:
Any reason for departing from the convention we've been using everywhere else of having a `class` to define a service, with a `setup` and `stop` method?

This matches the convention that's used in `plugin.ts` and also the way services are being authored in core & elsewhere in the data plugin.

Reply:
I was planning to do that, this is just basically `setup` method of `expressions` services.

Reply:
Done.